<!doctype html><html itemscope class=light lang=en-us itemtype=http://schema.org/WebPage><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1,maximum-scale=5"><meta name=theme-name content="hugoplate"><link rel="shortcut icon" href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png type=image/x-icon><link rel=icon href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png type=image/x-icon><link rel=icon type=image/png sizes=48x48 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_48x0_resize_lanczos_3.png><link rel=icon type=image/png sizes=96x96 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png><link rel=apple-touch-icon sizes=144x144 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_144x0_resize_lanczos_3.png><link rel=manifest href=/manifest.webmanifest><meta name=msapplication-TileColor content="#ddd"><meta name=theme-color content="#ffffff"><title>Bulk RNA-seq (8): Differential Analysis and Volcano Plotting with DESeq2</title>
<meta name=keywords content="Boilerplate,Hugo"><meta name=description content="this is meta description"><meta name=author content="zeon.studio"><meta property="og:image" content="/images/volcano.png"><meta name=twitter:image content="/images/volcano.png"><meta name=twitter:card content="summary
            _large_image
          "><meta property="og:image:width" content="14693"><meta property="og:image:height" content="9544"><meta property="og:image:type" content="image/
        .png
      "><meta property="og:title" content="Bulk RNA-seq (8): Differential Analysis and Volcano Plotting with DESeq2"><meta property="og:description" content="this is meta description"><meta property="og:type" content="website"><meta property="og:url" content="/blog/post-2/"><meta name=twitter:title content="Bulk RNA-seq (8): Differential Analysis and Volcano Plotting with DESeq2"><meta name=twitter:description content="this is meta description"><script>let indexURL="/searchindex.json",includeSectionsInSearch=["blog"],no_results_for="No results for",empty_search_results_placeholder="Type something to search.."</script><meta http-equiv=x-dns-prefetch-control content="on"><link rel=preconnect href=https://use.fontawesome.com crossorigin><link rel=preconnect href=//cdnjs.cloudflare.com><link rel=preconnect href=//www.googletagmanager.com><link rel=preconnect href=//www.google-analytics.com><link rel=dns-prefetch href=https://use.fontawesome.com><link rel=dns-prefetch href=//ajax.googleapis.com><link rel=dns-prefetch href=//cdnjs.cloudflare.com><link rel=dns-prefetch href=//www.googletagmanager.com><link rel=dns-prefetch href=//www.google-analytics.com><link rel=dns-prefetch href=//fonts.googleapis.com><link rel=dns-prefetch href=//connect.facebook.net><link rel=dns-prefetch href=//platform.linkedin.com><link rel=dns-prefetch href=//platform.twitter.com><link rel=preconnect href=https://fonts.googleapis.com><link rel=preconnect href=https://fonts.gstatic.com crossorigin><script>(function(){const e=document.createElement("link");e.href="https://fonts.googleapis.com/css2?family=Heebo:wght@400;600&family=Signika:wght@500;700&display=swap",e.type="text/css",e.rel="stylesheet",document.head.appendChild(e)})()</script><link href="/css/style.min.58d88361e6273a2d42f199a9d746b29f3a8c69c83bd057e830ca76269b1d6e2f.css" integrity="sha256-WNiDYeYnOi1C8Zmp10aynzqMacg70FfoMMp2Jpsdbi8=" rel=stylesheet><link defer async rel=stylesheet href="/css/style-lazy.min.f8b9be02754115291cc266be724b8382a0421c8040d5311fdac29495c8079c94.css" integrity="sha256-+Lm+AnVBFSkcwma+ckuDgqBCHIBA1TEf2sKUlcgHnJQ=" media=print onload='this.media="all",this.onload=null'></head><body><div class=preloader></div><script>(function(){"use strict";window.addEventListener("load",function(){document.querySelector(".preloader").style.display="none"})})()</script><header class="header sticky top-0 z-30"><nav class="navbar container"><div class=order-0><a class="navbar-brand block" href=/><img fetchpriority=high decoding=async class="img logo-light" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'>
<img fetchpriority=high decoding=async class="img logo-dark" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'></a></div><input id=nav-toggle type=checkbox class=hidden>
<label for=nav-toggle class="order-3 cursor-pointer flex items-center lg:hidden text-dark lg:order-1"><svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20"><title>Menu Open</title><path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0V0z"/></svg><svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20"><title>Menu Close</title><polygon points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2" transform="rotate(45 10 10)"/></svg></label><ul id=nav-menu class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8"><li class=nav-item><a class=nav-link href=/>Home</a></li><li class=nav-item><a class=nav-link href=/about/>About</a></li><li class=nav-item><a class=nav-link href=/blog/>Blog</a></li><li class=nav-item><a class=nav-link href=/categories/>Categories</a></li><li class="mt-4 inline-block lg:hidden"><a class="btn btn-outline-primary btn-sm" href=/contact>Contact me</a></li></ul><div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0"><button aria-label=search class="border-border text-dark hover:text-primary mr-5 inline-block border-r pr-5 text-xl" data-target=search-modal>
<i class="fa-solid fa-search"></i>
</button>
<a href=/contact class="btn btn-outline-primary btn-sm hidden lg:inline-block">Contact me</a></div></nav></header><div class=search-modal aria-hidden=true style=--color-primary:#121212><div data-target=close-search-modal class=search-modal-overlay></div><div class=search-wrapper data-image=true data-description=true data-tags=true data-categories=true><div class=search-wrapper-header><label for=search-modal-input style=margin-top:-1px><span class=sr-only>search icon</span>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" height="18" width="18" class="search-icon" data-type="search"><path fill="currentcolor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8.0 45.3s-32.8 12.5-45.3.0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9.0 208S93.1.0 208 0 416 93.1 416 208zM208 352a144 144 0 100-288 144 144 0 100 288z"/></svg>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" height="18" width="18" class="search-reset" data-type="reset"><path fill="currentcolor" d="M256 512A256 256 0 10256 0a256 256 0 100 512zM175 175c9.4-9.4 24.6-9.4 33.9.0l47 47 47-47c9.4-9.4 24.6-9.4 33.9.0s9.4 24.6.0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6.0 33.9s-24.6 9.4-33.9.0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9.0s-9.4-24.6.0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6.0-33.9z"/></svg>
</label><input id=search-modal-input type=text data-search-input autocomplete=off aria-label=Search placeholder="Search Post ..."></div><div class=search-wrapper-body><div class=search-result data-search-result></div><span class=search-result-empty>Type something to search..</span></div><div class=search-wrapper-footer><span><kbd><svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" fill="currentcolor" viewBox="0 0 16 16"><path d="M3.204 11h9.592L8 5.519 3.204 11zm-.753-.659 4.796-5.48a1 1 0 011.506.0l4.796 5.48c.566.647.106 1.659-.753 1.659H3.204a1 1 0 01-.753-1.659z"/></svg>
</kbd><kbd><svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" fill="currentcolor" style="margin-top:1px" viewBox="0 0 16 16"><path d="M3.204 5h9.592L8 10.481 3.204 5zm-.753.659 4.796 5.48a1 1 0 001.506.0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 00-.753 1.659z"/></svg>
</kbd>to navigate
</span><span><kbd><svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" fill="currentcolor" style="display:inline-block" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M14.5 1.5a.5.5.0 01.5.5v4.8a2.5 2.5.0 01-2.5 2.5H2.707l3.347 3.346a.5.5.0 01-.708.708l-4.2-4.2a.5.5.0 010-.708l4-4a.5.5.0 11.708.708L2.707 8.3H12.5A1.5 1.5.0 0014 6.8V2a.5.5.0 01.5-.5z"/></svg>
</kbd>to select
</span><span class=search-result-info></span>
<span data-target=close-search-modal><kbd>ESC</kbd> to close</span></div></div></div><main><section class="section pt-7"><div class=container><div class="row justify-center"><article class=lg:col-10><div class=mb-10><picture><source srcset=/images/volcano_hue3a76916b0cad22416e7870220f5c466_3075961_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/volcano_hue3a76916b0cad22416e7870220f5c466_3075961_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/volcano_hue3a76916b0cad22416e7870220f5c466_3075961_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/volcano_hue3a76916b0cad22416e7870220f5c466_3075961_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/volcano_hue3a76916b0cad22416e7870220f5c466_3075961_1110x0_resize_lanczos_3.png class="w-full rounded img" alt="Bulk RNA-seq (8): Differential Analysis and Volcano Plotting with DESeq2" width=14693 height=9544></picture></div><h1 class="h2 mb-4">Bulk RNA-seq (8): Differential Analysis and Volcano Plotting with DESeq2</h1><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-2"></i>
<a href=/categories/rna-seq/>Rna seq</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-clock mr-2"></i>
September 18, 2022</li></ul><div class="content mb-10"><div class="notice note"><div class=notice-head><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M10 9v5m0 5c-4.97056.0-9-4.0294-9-9 0-4.97056 4.02944-9 9-9 4.9706.0 9 4.02944 9 9 0 4.9706-4.0294 9-9 9zM10.0498 6V6.1L9.9502 6.1002V6H10.0498z" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/></svg><p>Note</p></div><div class=notice-body><p>Here, we demonstrate how to sift through the complex data to identify genes of interest and showcase their expression patterns through an elegant volcano plot. We&rsquo;ll also link back to our previous blog where we prepared the <code>ids</code> for gene annotation, tying our narrative together in a symphony of scientific discovery.</p></div></div><h3 id=differential-expression-analysis-with-deseq2>Differential Expression Analysis with DESeq2:</h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#75715e># Starting DESeq2 analysis to find differentially expressed genes (DEGs)</span>
</span></span><span style=display:flex><span>dds2 <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>DESeq</span>(dds)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Extracting and examining the DEGs</span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>resultsNames</span>(dds2)
</span></span><span style=display:flex><span>res <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>results</span>(dds2)
</span></span><span style=display:flex><span><span style=color:#a6e22e>head</span>(res)
</span></span><span style=display:flex><span><span style=color:#a6e22e>class</span>(res)
</span></span><span style=display:flex><span><span style=color:#a6e22e>summary</span>(res)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># The results are stored in a dataframe for easier data manipulation</span>
</span></span><span style=display:flex><span>res1 <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>data.frame</span>(res)
</span></span></code></pre></div><p><em>Explanation:</em> This section of the code initiates the differential expression analysis using the DESeq2 package and then extracts the results. These results are then converted into a dataframe for easier handling in subsequent steps.</p><h3 id=filtering-and-annotating-degs>Filtering and Annotating DEGs:</h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=display:flex><span><span style=color:#75715e># Filtering for significant DEGs</span>
</span></span><span style=display:flex><span>res_deseq2 <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>as.data.frame</span>(res) <span style=color:#f92672>%&gt;%</span>
</span></span><span style=display:flex><span>  <span style=color:#a6e22e>arrange</span>(padj) <span style=color:#f92672>%&gt;%</span>
</span></span><span style=display:flex><span>  dplyr<span style=color:#f92672>::</span><span style=color:#a6e22e>filter</span>(<span style=color:#a6e22e>abs</span>(log2FoldChange) <span style=color:#f92672>&gt;</span> <span style=color:#ae81ff>1</span>, padj <span style=color:#f92672>&lt;</span> <span style=color:#ae81ff>0.05</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Annotating DEGs based on their expression changes</span>
</span></span><span style=display:flex><span>res1 <span style=color:#f92672>%&gt;%</span>
</span></span><span style=display:flex><span>  <span style=color:#a6e22e>mutate</span>(group <span style=color:#f92672>=</span> <span style=color:#a6e22e>case_when</span>(
</span></span><span style=display:flex><span>    log2FoldChange <span style=color:#f92672>&gt;=</span> <span style=color:#ae81ff>1</span> <span style=color:#f92672>&amp;</span> padj <span style=color:#f92672>&lt;=</span> <span style=color:#ae81ff>0.05</span> <span style=color:#f92672>~</span> <span style=color:#e6db74>&#34;Up-regulated&#34;</span>,
</span></span><span style=display:flex><span>    log2FoldChange <span style=color:#f92672>&lt;=</span> <span style=color:#ae81ff>-1</span> <span style=color:#f92672>&amp;</span> padj <span style=color:#f92672>&lt;=</span> <span style=color:#ae81ff>0.05</span> <span style=color:#f92672>~</span> <span style=color:#e6db74>&#34;Down-regulated&#34;</span>,
</span></span><span style=display:flex><span>    <span style=color:#66d9ef>TRUE</span> <span style=color:#f92672>~</span> <span style=color:#e6db74>&#34;Not-significant&#34;</span>
</span></span><span style=display:flex><span>  )) <span style=color:#f92672>-&gt;</span> res2
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Summarizing the count of DEGs in each category</span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>table</span>(res2<span style=color:#f92672>$</span>group)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Exporting the annotated DEGs to a CSV file</span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>write.csv</span>(res_deseq2, file <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;diff_expr_results.csv&#34;</span>, quote <span style=color:#f92672>=</span> <span style=color:#66d9ef>FALSE</span>)
</span></span></code></pre></div><p><em>Explanation:</em> Here, the code filters the results to focus on genes with significant expression changes. The genes are then categorized as up-regulated, down-regulated, or not significant based on their log2 fold changes and adjusted p-values. A summary table is created to count the number of genes in each category, and the annotated data is saved as a CSV file.</p><h3 id=preparing-data-for-volcano-plotting>Preparing Data for Volcano Plotting:</h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-r data-lang=r><span style=display:flex><span><span style=color:#75715e># Preparing the data for visualization</span>
</span></span><span style=display:flex><span>deg.data <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>read.csv</span>(<span style=color:#e6db74>&#34;res2.csv&#34;</span>)
</span></span><span style=display:flex><span>deg.data<span style=color:#f92672>$</span>logp <span style=color:#f92672>&lt;-</span> <span style=color:#f92672>-</span><span style=color:#a6e22e>log10</span>(deg.data<span style=color:#f92672>$</span>padj)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Merging gene data with annotations from the previous blog post</span>
</span></span><span style=display:flex><span>gene_data <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>merge</span>(deg.data, ids, by <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;ensembl_gene_id&#34;</span>)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Adding labels for significant genes in the volcano plot</span>
</span></span><span style=display:flex><span>gene_data<span style=color:#f92672>$</span>label <span style=color:#f92672>&lt;-</span> <span style=color:#e6db74>&#34;&#34;</span>
</span></span><span style=display:flex><span>gene_data <span style=color:#f92672>&lt;-</span> gene_data<span style=color:#a6e22e>[order</span>(gene_data<span style=color:#f92672>$</span>log2FoldChange),]
</span></span><span style=display:flex><span>up.genes <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>tail</span>(gene_data<span style=color:#f92672>$</span>mgi_symbol<span style=color:#a6e22e>[which</span>(gene_data<span style=color:#f92672>$</span>group <span style=color:#f92672>==</span> <span style=color:#e6db74>&#34;Up-regulated&#34;</span>)], <span style=color:#ae81ff>5</span>)
</span></span><span style=display:flex><span>down.genes <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>head</span>(gene_data<span style=color:#f92672>$</span>mgi_symbol<span style=color:#a6e22e>[which</span>(gene_data<span style=color:#f92672>$</span>group <span style=color:#f92672>==</span> <span style=color:#e6db74>&#34;Down-regulated&#34;</span>)], <span style=color:#ae81ff>5</span>)
</span></span><span style=display:flex><span>gene_data<span style=color:#f92672>$</span>label<span style=color:#a6e22e>[match</span>(<span style=color:#a6e22e>c</span>(up.genes, down.genes), gene_data<span style=color:#f92672>$</span>mgi_symbol)] <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>c</span>(up.genes, down.genes)
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Removing any rows with missing data</span>
</span></span><span style=display:flex><span>gene_data <span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>na.omit</span>(gene_data)
</span></span></code></pre></div><p><em>Explanation:</em> The code snippet prepares the dataset for creating a volcano plot, a type of scatter plot that shows statistical significance (p-value) versus magnitude of change (fold change). The <code>ids</code> variable, which contains gene annotations from the previous blog post, is merged with the DEGs to annotate them properly. Significant genes are labeled for better visibility in the plot.</p><h3 id=creating-and-saving-the-volcano-plot>Creating and Saving the Volcano Plot:</h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-R data-lang=R><span style=display:flex><span><span style=color:#75715e># Creating the volcano plot using ggplot2</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>pi</span><span style=color:#f92672>&lt;-</span> <span style=color:#a6e22e>ggscatter</span>(gene_data, x<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;log2FoldChange&#34;</span>,y<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;logp&#34;</span>,
</span></span><span style=display:flex><span>               color<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;group&#34;</span>,
</span></span><span style=display:flex><span>               palette <span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#e6db74>&#34;#2F5688&#34;</span>,<span style=color:#e6db74>&#34;#BBBBBB&#34;</span>,<span style=color:#e6db74>&#34;#CC0000&#34;</span>),
</span></span><span style=display:flex><span>               size<span style=color:#f92672>=</span><span style=color:#ae81ff>3</span>,
</span></span><span style=display:flex><span>               alpha<span style=color:#f92672>=</span><span style=color:#ae81ff>0.5</span>,
</span></span><span style=display:flex><span>               repel<span style=color:#f92672>=</span>T,
</span></span><span style=display:flex><span>               xlim<span style=color:#f92672>=</span><span style=color:#a6e22e>c</span>(<span style=color:#ae81ff>-3</span>,<span style=color:#ae81ff>4</span>),
</span></span><span style=display:flex><span>               xlab <span style=color:#f92672>=</span> <span style=color:#e6db74>&#34;log2FoldChange&#34;</span>,
</span></span><span style=display:flex><span>               ylab <span style=color:#f92672>=</span><span style=color:#e6db74>&#34;-log10(Adjusted P-value)&#34;</span> ) <span style=color:#f92672>+</span>
</span></span><span style=display:flex><span>  <span style=color:#a6e22e>theme_base</span>() <span style=color:#f92672>+</span> 
</span></span><span style=display:flex><span>  <span style=color:#a6e22e>geom_hline</span>(yintercept<span style=color:#f92672>=</span><span style=color:#ae81ff>1.30</span>,linetype<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;dashed&#34;</span>)<span style=color:#f92672>+</span> 
</span></span><span style=display:flex><span>  <span style=color:#a6e22e>geom_vline</span>(xintercept<span style=color:#f92672>=</span> <span style=color:#a6e22e>c</span>(<span style=color:#ae81ff>-1</span>,<span style=color:#ae81ff>1</span>),linetype<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;dashed&#34;</span>) <span style=color:#f92672>+</span>
</span></span><span style=display:flex><span>  <span style=color:#a6e22e>geom_text_repel</span>(data<span style=color:#f92672>=</span><span style=color:#a6e22e>subset</span>(gene_data, <span style=color:#f92672>!</span><span style=color:#a6e22e>is.na</span>(label)), 
</span></span><span style=display:flex><span>                  <span style=color:#a6e22e>aes</span>(label<span style=color:#f92672>=</span>label), 
</span></span><span style=display:flex><span>                  size<span style=color:#f92672>=</span><span style=color:#ae81ff>3.5</span>, 
</span></span><span style=display:flex><span>                  box.padding <span style=color:#f92672>=</span> <span style=color:#a6e22e>unit</span>(<span style=color:#ae81ff>0.35</span>, <span style=color:#e6db74>&#34;lines&#34;</span>),
</span></span><span style=display:flex><span>                  point.padding <span style=color:#f92672>=</span> <span style=color:#a6e22e>unit</span>(<span style=color:#ae81ff>0.3</span>, <span style=color:#e6db74>&#34;lines&#34;</span>),
</span></span><span style=display:flex><span>                  max.overlaps <span style=color:#f92672>=</span> <span style=color:#ae81ff>100</span>)
</span></span><span style=display:flex><span><span style=color:#66d9ef>pi</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#a6e22e>ggsave</span>(<span style=color:#e6db74>&#34;volcano.png&#34;</span>,units<span style=color:#f92672>=</span><span style=color:#e6db74>&#34;cm&#34;</span>,dpi<span style=color:#f92672>=</span><span style=color:#ae81ff>1200</span>)
</span></span></code></pre></div><ul><li><code>ggscatter</code> is used to create a basic scatter plot with <code>log2FoldChange</code> on the x-axis and <code>-log10(Adjusted P-value)</code> on the y-axis.</li><li>The <code>color</code> argument, along with <code>palette</code>, is used to color-code points by their group, which represents whether they are upregulated, downregulated, or not significant.</li><li><code>geom_hline</code> and <code>geom_vline</code> add dashed lines to the plot, which help to visually identify the cutoffs for significance and fold change.</li><li><code>geom_text_repel</code> from the <code>ggrepel</code> package is used to add labels to significant genes, preventing them from overlapping and making them easier to read.</li></ul><p>Finally, we save the plot as a high-resolution PNG image:</p></div><div class="row items-start justify-between"><div class="lg:col-5 mb-10 flex items-center lg:mb-0"><h5 class=mr-3>Tags :</h5><ul><li class=inline-block><a class="bg-theme-light hover:bg-primary m-1 block rounded px-3 py-1 hover:text-white" href=/tags/rna-seq/>Rna seq</a></li><li class=inline-block><a class="bg-theme-light hover:bg-primary m-1 block rounded px-3 py-1 hover:text-white" href=/tags/scripts/>Scripts</a></li><li class=inline-block><a class="bg-theme-light hover:bg-primary m-1 block rounded px-3 py-1 hover:text-white" href=/tags/r/>R</a></li></ul></div><div class="lg:col-4 flex items-center"><div class=share-icons><h5 class=share-title>Share :</h5><a class="share-link share-facebook" href="https://facebook.com/sharer/sharer.php?u=%2fblog%2fpost-2%2f" target=_blank rel=noopener aria-label="share facebook"><span class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18.77 7.46H14.5v-1.9c0-.9.6-1.1 1-1.1h3V.5h-4.33C10.24.5 9.5 3.44 9.5 5.32v2.15h-3v4h3v12h5v-12h3.85l.42-4z"/></svg>
</span></a><a class="share-link share-twitter" href="https://twitter.com/intent/tweet/?text=Share&amp;url=%2fblog%2fpost-2%2f" target=_blank rel=noopener aria-label="share twitter"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8 2H1l8.26 11.015L1.45 22H4.1l6.388-7.349L16 22h7l-8.608-11.478L21.8 2h-2.65l-5.986 6.886zm9 18L5 4h2l12 16z"/></svg>
</span></a><a class="share-link share-email" href="mailto:?subject=Share&amp;body=%2fblog%2fpost-2%2f" target=_self rel=noopener aria-label="share email"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M22 4H2C.9 4 0 4.9.0 6v12c0 1.1.9 2 2 2h20c1.1.0 2-.9 2-2V6c0-1.1-.9-2-2-2zM7.25 14.43l-3.5 2c-.08.05-.17.07-.25.07-.17.0-.34-.1-.43-.25-.14-.24-.06-.55.18-.68l3.5-2c.24-.14.55-.06.68.18.14.24.06.55-.18.68zm4.75.07c-.1.0-.2-.03-.27-.08l-8.5-5.5c-.23-.15-.3-.46-.15-.7.15-.22.46-.3.7-.14L12 13.4l8.23-5.32c.23-.15.54-.08.7.15.14.23.07.54-.16.7l-8.5 5.5c-.08.04-.17.07-.27.07zm8.93 1.75c-.1.16-.26.25-.43.25-.08.0-.17-.02-.25-.07l-3.5-2c-.24-.13-.32-.44-.18-.68s.44-.32.68-.18l3.5 2c.24.13.32.44.18.68z"/></svg>
</span></a><a class="share-link share-reddit" href="https://reddit.com/submit/?url=%2fblog%2fpost-2%2f&amp;resubmit=true&amp;title=Share" target=_blank rel=noopener aria-label="share reddit"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M24 11.5c0-1.65-1.35-3-3-3-.96.0-1.86.48-2.42 1.24-1.64-1-3.75-1.64-6.07-1.72.08-1.1.4-3.05 1.52-3.7.72-.4 1.73-.24 3 .5C17.2 6.3 18.46 7.5 20 7.5c1.65.0 3-1.35 3-3s-1.35-3-3-3c-1.38.0-2.54.94-2.88 2.22-1.43-.72-2.64-.8-3.6-.25-1.64.94-1.95 3.47-2 4.55-2.33.08-4.45.7-6.1 1.72C4.86 8.98 3.96 8.5 3 8.5c-1.65.0-3 1.35-3 3 0 1.32.84 2.44 2.05 2.84-.03.22-.05.44-.05.66.0 3.86 4.5 7 10 7s10-3.14 10-7c0-.22-.02-.44-.05-.66 1.2-.4 2.05-1.54 2.05-2.84zM2.3 13.37C1.5 13.07 1 12.35 1 11.5c0-1.1.9-2 2-2 .64.0 1.22.32 1.6.82-1.1.85-1.92 1.9-2.3 3.05zm3.7.13c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2zm9.8 4.8c-1.08.63-2.42.96-3.8.96-1.4.0-2.74-.34-3.8-.95-.24-.13-.32-.44-.2-.68.15-.24.46-.32.7-.18 1.83 1.06 4.76 1.06 6.6.0.23-.13.53-.05.67.2.14.23.06.54-.18.67zm.2-2.8c-1.1.0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm5.7-2.13c-.38-1.16-1.2-2.2-2.3-3.05.38-.5.97-.82 1.6-.82 1.1.0 2 .9 2 2 0 .84-.53 1.57-1.3 1.87z"/></svg></span></a></div></div></div><div class=mt-20><div id=disqus_thread></div><script>window.disqus_config=function(){},function(){if(["localhost","127.0.0.1"].indexOf(window.location.hostname)!=-1){document.getElementById("disqus_thread").innerHTML="Disqus comments not available by default when the website is previewed locally.";return}var t=document,e=t.createElement("script");e.async=!0,e.src="//themefisher-template.disqus.com/embed.js",e.setAttribute("data-timestamp",+new Date),(t.head||t.body).appendChild(e)}()</script><noscript>Please enable JavaScript to view the <a href=https://disqus.com/?ref_noscript>comments powered by Disqus.</a></noscript><a href=https://disqus.com class=dsq-brlink>comments powered by <span class=logo-disqus>Disqus</span></a></div></article></div><div class="section pb-0"><h2 class="h3 mb-12">Related Posts</h2><div class=row><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="Bulk RNA-seq (6): Normalization and PCA Visualization in RNA-seq Data Analysis with DESeq2" width=9448 height=4724></picture><h4 class=mb-3><a href=/blog/normalization-and-pca-visualization-in-rna-seq-data-analysis-with-deseq2/>Bulk RNA-seq (6): Normalization and PCA Visualization in RNA-seq Data Analysis with DESeq2</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>Note
In this blog post, we&rsquo;ll dive into a practical guide on using DESeq2 for differential expression analysis in RNA-seq data.</p><a class="btn btn-outline-primary btn-sm" href=/blog/normalization-and-pca-visualization-in-rna-seq-data-analysis-with-deseq2/>Read More</a></div></div><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="bulk RNA-seq(2):Quality Control with FastQC and MultiQC" width=1200 height=800></picture><h4 class=mb-3><a href=/blog/qualitycheck/>bulk RNA-seq(2):Quality Control with FastQC and MultiQC</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>FastQC for Quality Checks FastQC provides a simple way to perform quality control checks on raw sequence data.</p><a class="btn btn-outline-primary btn-sm" href=/blog/qualitycheck/>Read More</a></div></div><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="Bulk RNA-seq (5): Streamlining Mapping with a Custom Linux Script" width=6000 height=3600></picture><h4 class=mb-3><a href=/blog/streamlining-mapping-with-a-custom-linux-script/>Bulk RNA-seq (5): Streamlining Mapping with a Custom Linux Script</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>Note
To overcome the inconvenience of manually mapping each RNA-seq sample to the reference genome, I&rsquo;ve developed a Linux shell script.</p><a class="btn btn-outline-primary btn-sm" href=/blog/streamlining-mapping-with-a-custom-linux-script/>Read More</a></div></div></div></div></div></section></main><footer class=bg-theme-light><div class=container><div class="row items-center py-10"><div class="lg:col-3 mb-8 text-center lg:mb-0 lg:text-left"><a class="navbar-brand inline-block" href=/><img fetchpriority=high decoding=async class="img logo-light" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'>
<img fetchpriority=high decoding=async class="img logo-dark" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'></a></div><div class="lg:col-6 mb-8 text-center lg:mb-0"><ul><li class="m-3 inline-block"><a href=/about/>About</a></li><li class="m-3 inline-block"><a href=/blog/>Blog</a></li></ul></div><div class="lg:col-3 mb-8 text-center lg:mb-0 lg:mt-0 lg:text-right"><ul class=social-icons><li><a target=_blank aria-label=facebook rel="nofollow noopener" href=https://www.facebook.com/><i class="fab fa-facebook"></i></a></li><li><a target=_blank aria-label=twitter rel="nofollow noopener" href=https://twitter.com/><i class="fab fa-twitter"></i></a></li><li><a target=_blank aria-label=github rel="nofollow noopener" href=https://www.github.com/><i class="fab fa-github"></i></a></li><li><a target=_blank aria-label=linkedin rel="nofollow noopener" href=https://www.linkedin.com/><i class="fab fa-linkedin"></i></a></li></ul></div></div></div><div class="border-border border-t py-7"><div class="text-light container text-center"><p>© 2018-2024 Yao Yuan. All rights reserved.</p></div></div></footer><script crossorigin=anonymous integrity="sha256-EjMwAhXn+vYO0J+UkUIy8BjNtBx1PoMmC+zQvUyDVsY=" src=/js/script.min.1233300215e7faf60ed09f94914232f018cdb41c753e83260becd0bd4c8356c6.js></script><script defer async crossorigin=anonymous integrity="sha256-9e9iyznGCSfFZB4ElW+WFonrJpHJr5pYxsaD7VVh4wc=" src=/js/script-lazy.min.f5ef62cb39c60927c5641e04956f961689eb2691c9af9a58c6c683ed5561e307.js></script><script>"serviceWorker"in navigator&&navigator.serviceWorker.register("/service-worker.js")</script></body></html>