<!doctype html><html itemscope class=light lang=en-us itemtype=http://schema.org/WebPage><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1,maximum-scale=5"><meta name=theme-name content="hugoplate"><link rel="shortcut icon" href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png type=image/x-icon><link rel=icon href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png type=image/x-icon><link rel=icon type=image/png sizes=48x48 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_48x0_resize_lanczos_3.png><link rel=icon type=image/png sizes=96x96 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png><link rel=apple-touch-icon sizes=144x144 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_144x0_resize_lanczos_3.png><link rel=manifest href=/manifest.webmanifest><meta name=msapplication-TileColor content="#ddd"><meta name=theme-color content="#ffffff"><title>What is the RNA-seq analysis?</title>
<meta name=keywords content="Boilerplate,Hugo"><meta name=description content="this is meta description"><meta name=author content="zeon.studio"><meta property="og:image" content="/images/RNA-seq.png"><meta name=twitter:image content="/images/RNA-seq.png"><meta name=twitter:card content="summary
            _large_image
          "><meta property="og:image:width" content="3600"><meta property="og:image:height" content="2100"><meta property="og:image:type" content="image/
        .png
      "><meta property="og:title" content="What is the RNA-seq analysis?"><meta property="og:description" content="this is meta description"><meta property="og:type" content="website"><meta property="og:url" content="/blog/what_is_rna_seq/"><meta name=twitter:title content="What is the RNA-seq analysis?"><meta name=twitter:description content="this is meta description"><script>let indexURL="/searchindex.json",includeSectionsInSearch=["blog"],no_results_for="No results for",empty_search_results_placeholder="Type something to search.."</script><meta http-equiv=x-dns-prefetch-control content="on"><link rel=preconnect href=https://use.fontawesome.com crossorigin><link rel=preconnect href=//cdnjs.cloudflare.com><link rel=preconnect href=//www.googletagmanager.com><link rel=preconnect href=//www.google-analytics.com><link rel=dns-prefetch href=https://use.fontawesome.com><link rel=dns-prefetch href=//ajax.googleapis.com><link rel=dns-prefetch href=//cdnjs.cloudflare.com><link rel=dns-prefetch href=//www.googletagmanager.com><link rel=dns-prefetch href=//www.google-analytics.com><link rel=dns-prefetch href=//fonts.googleapis.com><link rel=dns-prefetch href=//connect.facebook.net><link rel=dns-prefetch href=//platform.linkedin.com><link rel=dns-prefetch href=//platform.twitter.com><link rel=preconnect href=https://fonts.googleapis.com><link rel=preconnect href=https://fonts.gstatic.com crossorigin><script>(function(){const e=document.createElement("link");e.href="https://fonts.googleapis.com/css2?family=Heebo:wght@400;600&family=Signika:wght@500;700&display=swap",e.type="text/css",e.rel="stylesheet",document.head.appendChild(e)})()</script><link href="/css/style.min.58d88361e6273a2d42f199a9d746b29f3a8c69c83bd057e830ca76269b1d6e2f.css" integrity="sha256-WNiDYeYnOi1C8Zmp10aynzqMacg70FfoMMp2Jpsdbi8=" rel=stylesheet><link defer async rel=stylesheet href="/css/style-lazy.min.f8b9be02754115291cc266be724b8382a0421c8040d5311fdac29495c8079c94.css" integrity="sha256-+Lm+AnVBFSkcwma+ckuDgqBCHIBA1TEf2sKUlcgHnJQ=" media=print onload='this.media="all",this.onload=null'></head><body><div class=preloader></div><script>(function(){"use strict";window.addEventListener("load",function(){document.querySelector(".preloader").style.display="none"})})()</script><header class="header sticky top-0 z-30"><nav class="navbar container"><div class=order-0><a class="navbar-brand block" href=/><img fetchpriority=high decoding=async class="img logo-light" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'>
<img fetchpriority=high decoding=async class="img logo-dark" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'></a></div><input id=nav-toggle type=checkbox class=hidden>
<label for=nav-toggle class="order-3 cursor-pointer flex items-center lg:hidden text-dark lg:order-1"><svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20"><title>Menu Open</title><path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0V0z"/></svg><svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20"><title>Menu Close</title><polygon points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2" transform="rotate(45 10 10)"/></svg></label><ul id=nav-menu class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8"><li class=nav-item><a class=nav-link href=/>Home</a></li><li class=nav-item><a class=nav-link href=/about/>About</a></li><li class=nav-item><a class=nav-link href=/blog/>Blog</a></li><li class=nav-item><a class=nav-link href=/categories/>Categories</a></li><li class="mt-4 inline-block lg:hidden"><a class="btn btn-outline-primary btn-sm" href=/contact>Contact me</a></li></ul><div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0"><button aria-label=search class="border-border text-dark hover:text-primary mr-5 inline-block border-r pr-5 text-xl" data-target=search-modal>
<i class="fa-solid fa-search"></i>
</button>
<a href=/contact class="btn btn-outline-primary btn-sm hidden lg:inline-block">Contact me</a></div></nav></header><div class=search-modal aria-hidden=true style=--color-primary:#121212><div data-target=close-search-modal class=search-modal-overlay></div><div class=search-wrapper data-image=true data-description=true data-tags=true data-categories=true><div class=search-wrapper-header><label for=search-modal-input style=margin-top:-1px><span class=sr-only>search icon</span>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" height="18" width="18" class="search-icon" data-type="search"><path fill="currentcolor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8.0 45.3s-32.8 12.5-45.3.0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9.0 208S93.1.0 208 0 416 93.1 416 208zM208 352a144 144 0 100-288 144 144 0 100 288z"/></svg>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" height="18" width="18" class="search-reset" data-type="reset"><path fill="currentcolor" d="M256 512A256 256 0 10256 0a256 256 0 100 512zM175 175c9.4-9.4 24.6-9.4 33.9.0l47 47 47-47c9.4-9.4 24.6-9.4 33.9.0s9.4 24.6.0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6.0 33.9s-24.6 9.4-33.9.0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9.0s-9.4-24.6.0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6.0-33.9z"/></svg>
</label><input id=search-modal-input type=text data-search-input autocomplete=off aria-label=Search placeholder="Search Post ..."></div><div class=search-wrapper-body><div class=search-result data-search-result></div><span class=search-result-empty>Type something to search..</span></div><div class=search-wrapper-footer><span><kbd><svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" fill="currentcolor" viewBox="0 0 16 16"><path d="M3.204 11h9.592L8 5.519 3.204 11zm-.753-.659 4.796-5.48a1 1 0 011.506.0l4.796 5.48c.566.647.106 1.659-.753 1.659H3.204a1 1 0 01-.753-1.659z"/></svg>
</kbd><kbd><svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" fill="currentcolor" style="margin-top:1px" viewBox="0 0 16 16"><path d="M3.204 5h9.592L8 10.481 3.204 5zm-.753.659 4.796 5.48a1 1 0 001.506.0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 00-.753 1.659z"/></svg>
</kbd>to navigate
</span><span><kbd><svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" fill="currentcolor" style="display:inline-block" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M14.5 1.5a.5.5.0 01.5.5v4.8a2.5 2.5.0 01-2.5 2.5H2.707l3.347 3.346a.5.5.0 01-.708.708l-4.2-4.2a.5.5.0 010-.708l4-4a.5.5.0 11.708.708L2.707 8.3H12.5A1.5 1.5.0 0014 6.8V2a.5.5.0 01.5-.5z"/></svg>
</kbd>to select
</span><span class=search-result-info></span>
<span data-target=close-search-modal><kbd>ESC</kbd> to close</span></div></div></div><main><section class="section pt-7"><div class=container><div class="row justify-center"><article class=lg:col-10><div class=mb-10><picture><source srcset=/images/RNA-seq_hu4e571ce06a1b5f4636d562ec14a15535_411096_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/RNA-seq_hu4e571ce06a1b5f4636d562ec14a15535_411096_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/RNA-seq_hu4e571ce06a1b5f4636d562ec14a15535_411096_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/RNA-seq_hu4e571ce06a1b5f4636d562ec14a15535_411096_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/RNA-seq_hu4e571ce06a1b5f4636d562ec14a15535_411096_1110x0_resize_lanczos_3.png class="w-full rounded img" alt="What is the RNA-seq analysis?" width=3600 height=2100></picture></div><h1 class="h2 mb-4">What is the RNA-seq analysis?</h1><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-2"></i>
<a href=/categories/rna-seq/>Rna seq</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-clock mr-2"></i>
May 16, 2022</li></ul><div class="content mb-10"><h2 id=overall-description-of-rna-seq>Overall description of RNA-Seq</h2><p>The transcriptome has a high degree of complexity including multiple types of coding and non-coding RNA species, like mRNA, pre-mRNA, microRNA, and long ncRNA. To understand how the expression of a gene affects translation to proteins, <strong>messenger RNA (mRNA)</strong> molecules were the most frequently studied RNA species. Besides coding RNA, noncoding RNA molecules also play significant roles in cellular functions: <strong>Ribosomal RNAs</strong> and <strong>transfer RNAs</strong> are involved in mRNA translation. <strong>Small nuclear RNA (snRNAs)</strong> are involved in splicing. <strong>MicroRNA (miRNA)</strong> and <strong>piwi-interacting RNA (piRNA)</strong> regulate gene expression at the posttranscriptional level&mldr;</p><p>The development of high-throughput next-generation sequencing (NGS) has revolutionized transcriptomics by enabling RNA analysis through the sequencing of complementary DNA (cDNA). This method is termed RNA sequencing (RNA-Seq).</p><p>Understanding the Pandora box of the transcriptome is the key to connecting the information in our genome with its functional protein expression. RNA-seq analysis allows us to understand which genes are turned up/down in a cell/tissue, what their level of transcription is, and such changes-indicated diseases.</p><p>RNA-Seq can be applied to investigate different populations of RNA, including total RNA, pre-mRNA, and noncoding RNA, such as microRNA and long ncRNA.</p><p>An RNA-seq workflow has several steps, which can be summarized as:</p><ol><li><p>RNA extraction</p></li><li><p>Reverse transcription into cDNA (cDNA library preparation)</p></li><li><p>Adapted ligation</p></li><li><p>Amplification</p></li><li><p>Sequencing</p></li></ol><h3 id=1-rna-extraction>1. RNA extraction</h3><p>Overview of RNA-seq (<a href=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4863231/ target=_blank>Kimberly.KR et al.,<em>Cold Spring Harb Protoc.</em> 2015 </a>)</p><img src=https://s2.loli.net/2022/05/17/LDMnXhx8koCAy3p.jpg style=zoom:80%><p>The first step in transcriptome sequencing is the isolation of RNA from a biological sample as the above figure shows. To ensure a trustable RNA-seq experiment, the RNA needs to have sufficient quality to produce a library for sequencing.</p><p>Certain forms of contamination can be detected using UV spectroscopy and the calculation of absorbance ratios. Good quality RNA will have an A260/280 ratio of 1.7 to 2.1 and A260/230 of 2.0 to 2.2. A260/280 lower than 1.7 may indicate protein contamination, whereas A260/230 less than 1.8 may indicate the presence of organics (e.g. phenol or guanidine) leftover from the extraction protocol. Such contaminants may interfere with library preparation.</p><p>The measurement of RNA quality was usually performed by Agilent bioanalyzer: <strong>RNA Integrity Number (RIN)</strong>.</p><p><strong>RIN assigns an electropherogram a value of 1 to 10, with 10 being the least degraded, the best quality.</strong> RIN represents the ratios of 28S to 18S ribosomal bands. The figure below shows the ideal fluorescence-length graph of <strong>RIN 10</strong>.</p><p><img src=https://s2.loli.net/2022/05/17/cuLpQjKeX4vWTDs.png alt></p><div class="notice info"><div class=notice-head><svg width="20" height="20" viewBox="0 0 18 20" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M9.16109.993016C9.97971 1.03952 10.6611 1.42989 11.0721 2.22339l6.726 13.57801C18.4502 17.1739 17.4403 19.0208 15.7832 19.0474H2.23859C.730337 19.0234-.507163 17.3108.231587 15.7864L7.08321 2.20877c.12825-.24375.18675-.31425.29738-.44213.44475-.51562.93112-.791624 1.7805-.773624zM9.05046 2.49189C8.79284 2.50464 8.55696 2.64902 8.42834 2.87327 6.06134 7.36539 3.77946 11.9036 1.56546 16.4734 1.36071 16.9328 1.71209 17.5223 2.22621 17.547 6.74871 17.6201 11.2731 17.6201 15.7956 17.547 16.2925 17.523 16.666 16.953 16.459 16.4783c-2.1724-4.569-4.4119-9.10728-6.73729-13.60016C9.58446 2.63402 9.38309 2.48739 9.05046 2.49189z" fill="currentcolor"/><path d="M9.61323 13.2153H8.35773l-.138-6.16842h1.5375l-.144 6.16842zm-1.4355 1.8862C8.17773 14.8731 8.25161 14.6841 8.39973 14.5338 8.54823 14.3838 8.75036 14.3084 9.00648 14.3084 9.26298 14.3084 9.46511 14.3838 9.61323 14.5338 9.76136 14.6841 9.83561 14.8731 9.83561 15.1015 9.83561 15.3216 9.76323 15.5057 9.61923 15.6539 9.47486 15.802 9.27086 15.8762 9.00648 15.8762 8.74211 15.8762 8.53811 15.802 8.39373 15.6539 8.24973 15.5057 8.17773 15.3216 8.17773 15.1015z" fill="currentcolor"/></svg><p>Info</p></div><div class=notice-body><p>The total RNA ratio is calculated by taking the ratio of the area under the 18S and 28S rRNA peaks to the total area under the graph, a large number here is desired, indicating much of the rRNA is still at these sizes and thus little to no degradation has occurred. An ideal ratio can be seen in the above figure, where almost all of the RNA is in the 18S and 28S RNA peaks. A small marker height is desired, indicating only small amounts of RNA have been degraded and proceeded to the smallest lengths. If marker > 18S or 28S, meaning RNA has been greatly degraded.</p></div></div><p>Here is the sample RIN quality control graph. Most of the RINs are around 9, which means good quality of RNA.</p><p><img src=https://s2.loli.net/2022/05/17/9rvaUBdRm5HycjC.png alt></p><h3 id=2-library-preparation>2. Library preparation</h3><p>In the next step, we need to choose an appropriate library preparation protocol&mdash;&mdash;&mdash;-<strong>select RNA species!</strong></p><p>For eukaryotic systems, researchers have a choice between poly(A) selection or rRNA depletion for enriching non-ribosomal RNA during library preparation. Prokaryotes require rRNA depletion methods as their mRNA is not polyadenylated. The decision tree in Figure 10 can help guide the selection of an enrichment method. <a href=https://web.genewiz.com/hubfs/NGS/RNA-Seq/RNA-Seq%20eBook/NGS_RNA-Seq_eBook_US_Jul27_2020.pdf target=_blank>Reference</a></p><p><img src=https://s2.loli.net/2022/05/17/NTY19W5PXnDiZAH.png alt></p><p>Basically, if the rRNA transcripts are not removed before library construction, they will consume the bulk of the sequencing reads, thus limiting the detection of DEGs.</p><h4 id=21-polya-selection>2.1 Poly(A) selection</h4><p><img src=https://s2.loli.net/2022/05/18/hiy9HF2RVEJ8k6o.png alt></p><p>In a eukaryotic cell, 80% of the total RNAs are <strong>ribosomal RNA (rRNA)</strong>, whereas the other 5% is <strong>polyadenylated positive (poly(A)+ ) mRNA</strong>. Poly(A) selection is the most common method for library preparation to provide efficient enrichment of poly(A) + transcripts including mRNAs and many non-coding RNAs, and also reduces the amounts of pre-mRNAs. In this approach, the 3′ poly-A tail of mRNA molecules is targeted using poly-T oligos that are covalently attached to a given substrate (e.g., magnetic beads).</p><h4 id=22-rrna-depletion>2.2 rRNA depletion</h4><p>rRNA-depleted removes cytoplasmic and mitochondrial rRNA and thus includes poly(A) + mRNA, as well as non-coding RNAs or protein-coding mRNAs that are not polyadenylated. In eukaryotes, this technique allows researchers to study both coding and non-coding transcripts.</p><p>In a previous study, the researchers showed that rRNA depletion captured more unique transcriptome features, whereas polyA+ selection outperformed rRNA depletion with higher exonic coverage and better accuracy of gene quantification. <a href=https://www.nature.com/articles/s41598-018-23226-4#Sec1 target=_blank>Reference</a></p><p><img src=https://s2.loli.net/2022/05/18/aC7RhH3AQT19XkS.png alt></p><h3 id=3-sequencing>3. Sequencing</h3><p><img src=https://s2.loli.net/2022/05/19/bYvAC8VoNaIk1EB.jpg alt></p><p>In RNA-seq sequencing, there are paired-end sequencing and single-end sequencing.</p><p>Paired-end sequencing is usually recommended for most applications as it provides richer data and permits longer library insert sizes. Paired-end sequencing allows us to sequence both ends of a fragment and generate high-quality, alignable sequence data.</p><p>For a single-read run, one Read 1 (R1) FASTQ file is created for each sample per flow cell lane. For a paired-end run, one R1 and one Read 2 (R2) FASTQ file is created for each sample for each lane. FASTQ files are compressed and created with the extension *<em>.fastq.gz.</em></p><p>Currently, the Illumina HiSeq platform is the most commonly applied next-generation sequencing technology for RNA-Seq and has set the standard for NGS sequencing. The platform has two flow cells, each providing eight separate lanes for sequencing reactions to occur.</p><p>Normally, we will get 8 read files (.fastq) for one sample. Each sample run in 4 separate lanes with paired-end:</p><p>XXX_L001_R1_001.fastq</p><p>XXX_L002_R1_001.fastq</p><p>XXX_L003_R1_001.fastq</p><p>XXX_L004_R1_001.fastq</p><p>XXX_L001_R2_001.fastq</p><p>XXX_L002_R2_001.fastq</p><p>XXX_L003_R2_001.fastq</p><p>XXX_L004_R2_001.fastq</p><p>So, we need to <code>cat</code>the files together into one.</p><h3 id=4-rna-seq-data-analysis>4. RNA-seq data analysis</h3><p><img src=https://s2.loli.net/2022/05/19/IxYKP2uh3QFA7eC.png alt></p><p>My RNA-seq analysis workflow is based on this figure. For more information, please refer to my other articles.</p></div><div class="row items-start justify-between"><div class="lg:col-5 mb-10 flex items-center lg:mb-0"><h5 class=mr-3>Tags :</h5><ul><li class=inline-block><a class="bg-theme-light hover:bg-primary m-1 block rounded px-3 py-1 hover:text-white" href=/tags/rna-seq/>Rna seq</a></li></ul></div><div class="lg:col-4 flex items-center"><div class=share-icons><h5 class=share-title>Share :</h5><a class="share-link share-facebook" href="https://facebook.com/sharer/sharer.php?u=%2fblog%2fwhat_is_rna_seq%2f" target=_blank rel=noopener aria-label="share facebook"><span class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18.77 7.46H14.5v-1.9c0-.9.6-1.1 1-1.1h3V.5h-4.33C10.24.5 9.5 3.44 9.5 5.32v2.15h-3v4h3v12h5v-12h3.85l.42-4z"/></svg>
</span></a><a class="share-link share-twitter" href="https://twitter.com/intent/tweet/?text=Share&amp;url=%2fblog%2fwhat_is_rna_seq%2f" target=_blank rel=noopener aria-label="share twitter"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8 2H1l8.26 11.015L1.45 22H4.1l6.388-7.349L16 22h7l-8.608-11.478L21.8 2h-2.65l-5.986 6.886zm9 18L5 4h2l12 16z"/></svg>
</span></a><a class="share-link share-email" href="mailto:?subject=Share&amp;body=%2fblog%2fwhat_is_rna_seq%2f" target=_self rel=noopener aria-label="share email"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M22 4H2C.9 4 0 4.9.0 6v12c0 1.1.9 2 2 2h20c1.1.0 2-.9 2-2V6c0-1.1-.9-2-2-2zM7.25 14.43l-3.5 2c-.08.05-.17.07-.25.07-.17.0-.34-.1-.43-.25-.14-.24-.06-.55.18-.68l3.5-2c.24-.14.55-.06.68.18.14.24.06.55-.18.68zm4.75.07c-.1.0-.2-.03-.27-.08l-8.5-5.5c-.23-.15-.3-.46-.15-.7.15-.22.46-.3.7-.14L12 13.4l8.23-5.32c.23-.15.54-.08.7.15.14.23.07.54-.16.7l-8.5 5.5c-.08.04-.17.07-.27.07zm8.93 1.75c-.1.16-.26.25-.43.25-.08.0-.17-.02-.25-.07l-3.5-2c-.24-.13-.32-.44-.18-.68s.44-.32.68-.18l3.5 2c.24.13.32.44.18.68z"/></svg>
</span></a><a class="share-link share-reddit" href="https://reddit.com/submit/?url=%2fblog%2fwhat_is_rna_seq%2f&amp;resubmit=true&amp;title=Share" target=_blank rel=noopener aria-label="share reddit"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M24 11.5c0-1.65-1.35-3-3-3-.96.0-1.86.48-2.42 1.24-1.64-1-3.75-1.64-6.07-1.72.08-1.1.4-3.05 1.52-3.7.72-.4 1.73-.24 3 .5C17.2 6.3 18.46 7.5 20 7.5c1.65.0 3-1.35 3-3s-1.35-3-3-3c-1.38.0-2.54.94-2.88 2.22-1.43-.72-2.64-.8-3.6-.25-1.64.94-1.95 3.47-2 4.55-2.33.08-4.45.7-6.1 1.72C4.86 8.98 3.96 8.5 3 8.5c-1.65.0-3 1.35-3 3 0 1.32.84 2.44 2.05 2.84-.03.22-.05.44-.05.66.0 3.86 4.5 7 10 7s10-3.14 10-7c0-.22-.02-.44-.05-.66 1.2-.4 2.05-1.54 2.05-2.84zM2.3 13.37C1.5 13.07 1 12.35 1 11.5c0-1.1.9-2 2-2 .64.0 1.22.32 1.6.82-1.1.85-1.92 1.9-2.3 3.05zm3.7.13c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2zm9.8 4.8c-1.08.63-2.42.96-3.8.96-1.4.0-2.74-.34-3.8-.95-.24-.13-.32-.44-.2-.68.15-.24.46-.32.7-.18 1.83 1.06 4.76 1.06 6.6.0.23-.13.53-.05.67.2.14.23.06.54-.18.67zm.2-2.8c-1.1.0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm5.7-2.13c-.38-1.16-1.2-2.2-2.3-3.05.38-.5.97-.82 1.6-.82 1.1.0 2 .9 2 2 0 .84-.53 1.57-1.3 1.87z"/></svg></span></a></div></div></div><div class=mt-20><div id=disqus_thread></div><script>window.disqus_config=function(){},function(){if(["localhost","127.0.0.1"].indexOf(window.location.hostname)!=-1){document.getElementById("disqus_thread").innerHTML="Disqus comments not available by default when the website is previewed locally.";return}var t=document,e=t.createElement("script");e.async=!0,e.src="//themefisher-template.disqus.com/embed.js",e.setAttribute("data-timestamp",+new Date),(t.head||t.body).appendChild(e)}()</script><noscript>Please enable JavaScript to view the <a href=https://disqus.com/?ref_noscript>comments powered by Disqus.</a></noscript><a href=https://disqus.com class=dsq-brlink>comments powered by <span class=logo-disqus>Disqus</span></a></div></article></div><div class="section pb-0"><h2 class="h3 mb-12">Related Posts</h2><div class=row><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/multiqc_huefd6b0aaaa97d45f9ad7329e016c40fd_77022_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="bulk RNA-seq(2):Quality Control with FastQC and MultiQC" width=1200 height=800></picture><h4 class=mb-3><a href=/blog/qualitycheck/>bulk RNA-seq(2):Quality Control with FastQC and MultiQC</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>FastQC for Quality Checks FastQC provides a simple way to perform quality control checks on raw sequence data.</p><a class="btn btn-outline-primary btn-sm" href=/blog/qualitycheck/>Read More</a></div></div><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/concatenation_hu7249349a9a5ec8067785a059cd81390a_19293_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/concatenation_hu7249349a9a5ec8067785a059cd81390a_19293_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/concatenation_hu7249349a9a5ec8067785a059cd81390a_19293_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/concatenation_hu7249349a9a5ec8067785a059cd81390a_19293_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/concatenation_hu7249349a9a5ec8067785a059cd81390a_19293_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="bulk RNA-seq(1):Concatenation of raw read files" width=737 height=191></picture><h4 class=mb-3><a href=/blog/rna-seq1concatenation/>bulk RNA-seq(1):Concatenation of raw read files</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>Concatenation of Raw Read Files To begin with, my operating environment is Linux Ubuntu 20.</p><a class="btn btn-outline-primary btn-sm" href=/blog/rna-seq1concatenation/>Read More</a></div></div><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/bulk7_hua3125c79ca99759241aa1892e3a0e319_24526_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/bulk7_hua3125c79ca99759241aa1892e3a0e319_24526_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/bulk7_hua3125c79ca99759241aa1892e3a0e319_24526_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/bulk7_hua3125c79ca99759241aa1892e3a0e319_24526_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/bulk7_hua3125c79ca99759241aa1892e3a0e319_24526_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="Bulk RNA-seq (7): From ENSEMBL ID Cleanup to Gene Name Conversion" width=967 height=266></picture><h4 class=mb-3><a href=/blog/from-ensembl-id-cleanup-to-gene-name-conversion/>Bulk RNA-seq (7): From ENSEMBL ID Cleanup to Gene Name Conversion</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>Note
In the realm of RNA-seq analysis, one often encounters the challenge of working with ENSEMBL IDs that include decimal points, a result of gene versioning in databases.</p><a class="btn btn-outline-primary btn-sm" href=/blog/from-ensembl-id-cleanup-to-gene-name-conversion/>Read More</a></div></div></div></div></div></section></main><footer class=bg-theme-light><div class=container><div class="row items-center py-10"><div class="lg:col-3 mb-8 text-center lg:mb-0 lg:text-left"><a class="navbar-brand inline-block" href=/><img fetchpriority=high decoding=async class="img logo-light" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'>
<img fetchpriority=high decoding=async class="img logo-dark" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'></a></div><div class="lg:col-6 mb-8 text-center lg:mb-0"><ul><li class="m-3 inline-block"><a href=/about/>About</a></li><li class="m-3 inline-block"><a href=/blog/>Blog</a></li></ul></div><div class="lg:col-3 mb-8 text-center lg:mb-0 lg:mt-0 lg:text-right"><ul class=social-icons><li><a target=_blank aria-label=facebook rel="nofollow noopener" href=https://www.facebook.com/><i class="fab fa-facebook"></i></a></li><li><a target=_blank aria-label=twitter rel="nofollow noopener" href=https://twitter.com/><i class="fab fa-twitter"></i></a></li><li><a target=_blank aria-label=github rel="nofollow noopener" href=https://www.github.com/><i class="fab fa-github"></i></a></li><li><a target=_blank aria-label=linkedin rel="nofollow noopener" href=https://www.linkedin.com/><i class="fab fa-linkedin"></i></a></li></ul></div></div></div><div class="border-border border-t py-7"><div class="text-light container text-center"><p>© 2018-2024 Yao Yuan. All rights reserved.</p></div></div></footer><script crossorigin=anonymous integrity="sha256-EjMwAhXn+vYO0J+UkUIy8BjNtBx1PoMmC+zQvUyDVsY=" src=/js/script.min.1233300215e7faf60ed09f94914232f018cdb41c753e83260becd0bd4c8356c6.js></script><script defer async crossorigin=anonymous integrity="sha256-9e9iyznGCSfFZB4ElW+WFonrJpHJr5pYxsaD7VVh4wc=" src=/js/script-lazy.min.f5ef62cb39c60927c5641e04956f961689eb2691c9af9a58c6c683ed5561e307.js></script><script>"serviceWorker"in navigator&&navigator.serviceWorker.register("/service-worker.js")</script></body></html>