<!doctype html><html itemscope class=light lang=en-us itemtype=http://schema.org/WebPage><head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1,maximum-scale=5"><meta name=theme-name content="hugoplate"><link rel="shortcut icon" href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png type=image/x-icon><link rel=icon href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png type=image/x-icon><link rel=icon type=image/png sizes=48x48 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_48x0_resize_lanczos_3.png><link rel=icon type=image/png sizes=96x96 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_96x0_resize_lanczos_3.png><link rel=apple-touch-icon sizes=144x144 href=/images/favicon1_hu1ab4944a393ddd850c1139725afbb205_17560_144x0_resize_lanczos_3.png><link rel=manifest href=/manifest.webmanifest><meta name=msapplication-TileColor content="#ddd"><meta name=theme-color content="#ffffff"><title>bulk RNA-seq(3):Trim the read files using trim_galore</title>
<meta name=keywords content="Boilerplate,Hugo"><meta name=description content="this is meta description"><meta name=author content="zeon.studio"><meta property="og:image" content="/images/trim2.png"><meta name=twitter:image content="/images/trim2.png"><meta name=twitter:card content="summary
            _large_image
          "><meta property="og:image:width" content="6000"><meta property="og:image:height" content="3600"><meta property="og:image:type" content="image/
        .png
      "><meta property="og:title" content="bulk RNA-seq(3):Trim the read files using trim_galore"><meta property="og:description" content="this is meta description"><meta property="og:type" content="website"><meta property="og:url" content="/blog/trim-the-read-files-using-trim_galore/"><meta name=twitter:title content="bulk RNA-seq(3):Trim the read files using trim_galore"><meta name=twitter:description content="this is meta description"><script>let indexURL="/searchindex.json",includeSectionsInSearch=["blog"],no_results_for="No results for",empty_search_results_placeholder="Type something to search.."</script><meta http-equiv=x-dns-prefetch-control content="on"><link rel=preconnect href=https://use.fontawesome.com crossorigin><link rel=preconnect href=//cdnjs.cloudflare.com><link rel=preconnect href=//www.googletagmanager.com><link rel=preconnect href=//www.google-analytics.com><link rel=dns-prefetch href=https://use.fontawesome.com><link rel=dns-prefetch href=//ajax.googleapis.com><link rel=dns-prefetch href=//cdnjs.cloudflare.com><link rel=dns-prefetch href=//www.googletagmanager.com><link rel=dns-prefetch href=//www.google-analytics.com><link rel=dns-prefetch href=//fonts.googleapis.com><link rel=dns-prefetch href=//connect.facebook.net><link rel=dns-prefetch href=//platform.linkedin.com><link rel=dns-prefetch href=//platform.twitter.com><link rel=preconnect href=https://fonts.googleapis.com><link rel=preconnect href=https://fonts.gstatic.com crossorigin><script>(function(){const e=document.createElement("link");e.href="https://fonts.googleapis.com/css2?family=Heebo:wght@400;600&family=Signika:wght@500;700&display=swap",e.type="text/css",e.rel="stylesheet",document.head.appendChild(e)})()</script><link href="/css/style.min.58d88361e6273a2d42f199a9d746b29f3a8c69c83bd057e830ca76269b1d6e2f.css" integrity="sha256-WNiDYeYnOi1C8Zmp10aynzqMacg70FfoMMp2Jpsdbi8=" rel=stylesheet><link defer async rel=stylesheet href="/css/style-lazy.min.f8b9be02754115291cc266be724b8382a0421c8040d5311fdac29495c8079c94.css" integrity="sha256-+Lm+AnVBFSkcwma+ckuDgqBCHIBA1TEf2sKUlcgHnJQ=" media=print onload='this.media="all",this.onload=null'></head><body><div class=preloader></div><script>(function(){"use strict";window.addEventListener("load",function(){document.querySelector(".preloader").style.display="none"})})()</script><header class="header sticky top-0 z-30"><nav class="navbar container"><div class=order-0><a class="navbar-brand block" href=/><img fetchpriority=high decoding=async class="img logo-light" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'>
<img fetchpriority=high decoding=async class="img logo-dark" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'></a></div><input id=nav-toggle type=checkbox class=hidden>
<label for=nav-toggle class="order-3 cursor-pointer flex items-center lg:hidden text-dark lg:order-1"><svg id="show-button" class="h-6 fill-current block" viewBox="0 0 20 20"><title>Menu Open</title><path d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0V0z"/></svg><svg id="hide-button" class="h-6 fill-current hidden" viewBox="0 0 20 20"><title>Menu Close</title><polygon points="11 9 22 9 22 11 11 11 11 22 9 22 9 11 -2 11 -2 9 9 9 9 -2 11 -2" transform="rotate(45 10 10)"/></svg></label><ul id=nav-menu class="navbar-nav order-3 hidden lg:flex w-full pb-6 lg:order-1 lg:w-auto lg:space-x-2 lg:pb-0 xl:space-x-8"><li class=nav-item><a class=nav-link href=/>Home</a></li><li class=nav-item><a class=nav-link href=/about/>About</a></li><li class=nav-item><a class=nav-link href=/blog/>Blog</a></li><li class=nav-item><a class=nav-link href=/categories/>Categories</a></li><li class="mt-4 inline-block lg:hidden"><a class="btn btn-outline-primary btn-sm" href=/contact>Contact me</a></li></ul><div class="order-1 ml-auto flex items-center md:order-2 lg:ml-0"><button aria-label=search class="border-border text-dark hover:text-primary mr-5 inline-block border-r pr-5 text-xl" data-target=search-modal>
<i class="fa-solid fa-search"></i>
</button>
<a href=/contact class="btn btn-outline-primary btn-sm hidden lg:inline-block">Contact me</a></div></nav></header><div class=search-modal aria-hidden=true style=--color-primary:#121212><div data-target=close-search-modal class=search-modal-overlay></div><div class=search-wrapper data-image=true data-description=true data-tags=true data-categories=true><div class=search-wrapper-header><label for=search-modal-input style=margin-top:-1px><span class=sr-only>search icon</span>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" height="18" width="18" class="search-icon" data-type="search"><path fill="currentcolor" d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8.0 45.3s-32.8 12.5-45.3.0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9.0 208S93.1.0 208 0 416 93.1 416 208zM208 352a144 144 0 100-288 144 144 0 100 288z"/></svg>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" height="18" width="18" class="search-reset" data-type="reset"><path fill="currentcolor" d="M256 512A256 256 0 10256 0a256 256 0 100 512zM175 175c9.4-9.4 24.6-9.4 33.9.0l47 47 47-47c9.4-9.4 24.6-9.4 33.9.0s9.4 24.6.0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6.0 33.9s-24.6 9.4-33.9.0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9.0s-9.4-24.6.0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6.0-33.9z"/></svg>
</label><input id=search-modal-input type=text data-search-input autocomplete=off aria-label=Search placeholder="Search Post ..."></div><div class=search-wrapper-body><div class=search-result data-search-result></div><span class=search-result-empty>Type something to search..</span></div><div class=search-wrapper-footer><span><kbd><svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" fill="currentcolor" viewBox="0 0 16 16"><path d="M3.204 11h9.592L8 5.519 3.204 11zm-.753-.659 4.796-5.48a1 1 0 011.506.0l4.796 5.48c.566.647.106 1.659-.753 1.659H3.204a1 1 0 01-.753-1.659z"/></svg>
</kbd><kbd><svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" fill="currentcolor" style="margin-top:1px" viewBox="0 0 16 16"><path d="M3.204 5h9.592L8 10.481 3.204 5zm-.753.659 4.796 5.48a1 1 0 001.506.0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 00-.753 1.659z"/></svg>
</kbd>to navigate
</span><span><kbd><svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" fill="currentcolor" style="display:inline-block" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M14.5 1.5a.5.5.0 01.5.5v4.8a2.5 2.5.0 01-2.5 2.5H2.707l3.347 3.346a.5.5.0 01-.708.708l-4.2-4.2a.5.5.0 010-.708l4-4a.5.5.0 11.708.708L2.707 8.3H12.5A1.5 1.5.0 0014 6.8V2a.5.5.0 01.5-.5z"/></svg>
</kbd>to select
</span><span class=search-result-info></span>
<span data-target=close-search-modal><kbd>ESC</kbd> to close</span></div></div></div><main><section class="section pt-7"><div class=container><div class="row justify-center"><article class=lg:col-10><div class=mb-10><picture><source srcset=/images/trim2_hu66a195306e7d03dd8abb316f1d6fa6b6_1243274_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/trim2_hu66a195306e7d03dd8abb316f1d6fa6b6_1243274_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/trim2_hu66a195306e7d03dd8abb316f1d6fa6b6_1243274_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/trim2_hu66a195306e7d03dd8abb316f1d6fa6b6_1243274_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/trim2_hu66a195306e7d03dd8abb316f1d6fa6b6_1243274_1110x0_resize_lanczos_3.png class="w-full rounded img" alt="bulk RNA-seq(3):Trim the read files using trim_galore" width=6000 height=3600></picture></div><h1 class="h2 mb-4">bulk RNA-seq(3):Trim the read files using trim_galore</h1><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-2"></i>
<a href=/categories/rna-seq/>Rna seq</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-clock mr-2"></i>
May 30, 2022</li></ul><div class="content mb-10"><div class="notice tip"><div class=notice-head><svg width="20" height="20" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path fillRule="evenodd" clipRule="evenodd" d="M12 0c6.6274.0 12 5.37258 12 12 0 6.6274-5.3726 12-12 12C5.37258 24 0 18.6274.0 12 0 5.37258 5.37258.0 12 0zm0 2.4C6.69807 2.4 2.4 6.69807 2.4 12c0 5.3019 4.29807 9.6 9.6 9.6 5.3019.0 9.6-4.2981 9.6-9.6.0-5.30193-4.2981-9.6-9.6-9.6zm3.9515 5.15147L9.6 13.9029 8.04853 12.3515C7.5799 11.8828 6.8201 11.8828 6.35147 12.3515c-.46863.4686-.46863 1.2284.0 1.697l2.4 2.4C9.2201 16.9172 9.9799 16.9172 10.4485 16.4485l7.2-7.19997C18.1172 8.7799 18.1172 8.0201 17.6485 7.55147c-.468599999999999-.46863-1.2284-.46863-1.697.0z" fill="currentcolor"/></svg><p>Tip</p></div><div class=notice-body><p>We will install <code>Trim Galore</code> and trim the merged read files to improve the quality of our data.</p></div></div><h2 id=1-the-importance-of-trimming>1. The Importance of Trimming</h2><p>Quality trimming reduces the overall number of reads but increases the total and proportion of uniquely mapped reads, yielding more useful data for downstream analyses. However, overly aggressive trimming can adversely affect subsequent analyses.</p><p>:man_teacher: Trimming removes low-quality reads! It&rsquo;s essential before mapping!</p><p>:woman_teacher: Be cautious! Trimming might remove critical reads, including those considered low quality.</p><p>:bulb: <strong>The decision to trim should be based on your project needs and <code>FastQC</code> results.</strong></p><h2 id=2-assessing-read-quality-with-fastqc>2. Assessing Read Quality with <code>FastQC</code></h2><p>Refer to my previous article for how to use <code>FastQC</code>.</p><h3 id=21-per-base-sequence-quality>2.1 Per Base Sequence Quality</h3><img src=https://s2.loli.net/2022/04/08/IWVoEHvpxZ4L2yT.png style=zoom:90%><p>:warning: The figure above shows that the last 10 base pairs are of low quality, especially the very last one (marked with a red circle). Therefore, I&rsquo;ve decided to trim this sample.</p><h2 id=3-trimming-with-trim-galore>3. Trimming with <code>Trim Galore</code></h2><div class="notice note"><div class=notice-head><svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M10 9v5m0 5c-4.97056.0-9-4.0294-9-9 0-4.97056 4.02944-9 9-9 4.9706.0 9 4.02944 9 9 0 4.9706-4.0294 9-9 9zM10.0498 6V6.1L9.9502 6.1002V6H10.0498z" stroke="currentcolor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/></svg><p>Note</p></div><div class=notice-body><p>Our guide follows the <a href=https://github.com/FelixKrueger/TrimGalore/blob/master/Docs/Trim_Galore_User_Guide.md target=_blank>Trim Galore instructions</a>
.</p></div></div><h3 id=31-installing-trim-galore>3.1 Installing <code>Trim Galore</code></h3><p>Visit the <a href=https://github.com/FelixKrueger/TrimGalore/releases target=_blank>download site</a>
on GitHub.</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-shell data-lang=shell><span style=display:flex><span><span style=color:#75715e># Download the trim galore tar.gz file</span>
</span></span><span style=display:flex><span>tar -zxvf 0.6.7.tar.gz
</span></span></code></pre></div><p>Verify the installation:</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-shell data-lang=shell><span style=display:flex><span>trim_galore -version
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>trim_galore --help
</span></span></code></pre></div><h3 id=32-running-trim-galore>3.2 Running <code>Trim Galore</code></h3><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-shell data-lang=shell><span style=display:flex><span>trim_galore -q <span style=color:#ae81ff>25</span> --phred <span style=color:#ae81ff>33</span> --stringency <span style=color:#ae81ff>3</span> --length <span style=color:#ae81ff>30</span> --paired 1_R1.fastq.gz 1_R2.fastq.gz --gzip -o./cleandata
</span></span></code></pre></div><p>Trimming a paired sample typically takes about 15 minutes.</p><h3 id=33-automating-trimming-with-a-shell-script>3.3 Automating Trimming with a Shell Script</h3><p>I have created a shell script to automate the trimming process with Trim Galore. This script is designed to trim paired-end sequencing files for a series of samples by specifying their prefixes.</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-shell data-lang=shell><span style=display:flex><span><span style=color:#75715e>#!/bin/bash
</span></span></span><span style=display:flex><span><span style=color:#75715e></span>
</span></span><span style=display:flex><span><span style=color:#75715e># Specify the output directory where trimmed files will be stored</span>
</span></span><span style=display:flex><span>OUT_DIR<span style=color:#f92672>=</span>~/OUT_DIR
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># List of sample prefixes to process</span>
</span></span><span style=display:flex><span>Prefixes<span style=color:#f92672>=(</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_13&#34;</span> 
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_14&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_15&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_16&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_17&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_18&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_19&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_20&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_21&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_22&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_23&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_24&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_25&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_26&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_27&#34;</span>
</span></span><span style=display:flex><span><span style=color:#e6db74>&#34;LAB_410_28&#34;</span>
</span></span><span style=display:flex><span><span style=color:#f92672>)</span>
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span><span style=color:#75715e># Loop through each prefix to construct file names for R1 and R2 reads and perform trimming</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>for</span> PREFIX in <span style=color:#e6db74>&#34;</span><span style=color:#e6db74>${</span>Prefixes[@]<span style=color:#e6db74>}</span><span style=color:#e6db74>&#34;</span>
</span></span><span style=display:flex><span><span style=color:#66d9ef>do</span>
</span></span><span style=display:flex><span>  R1<span style=color:#f92672>=</span><span style=color:#e6db74>${</span>PREFIX<span style=color:#e6db74>}</span>_R1.fastq.gz
</span></span><span style=display:flex><span>  R2<span style=color:#f92672>=</span><span style=color:#e6db74>${</span>PREFIX<span style=color:#e6db74>}</span>_R2.fastq.gz
</span></span><span style=display:flex><span>
</span></span><span style=display:flex><span>  <span style=color:#75715e># Execute Trim Galore with specified parameters</span>
</span></span><span style=display:flex><span>  trim_galore -q <span style=color:#ae81ff>25</span> --phred33 --stringency <span style=color:#ae81ff>3</span> --length <span style=color:#ae81ff>30</span> --paired $R1 $R2 --gzip -o $OUT_DIR
</span></span><span style=display:flex><span><span style=color:#66d9ef>done</span>
</span></span></code></pre></div><p><strong>Script Explanation and Usage:</strong></p><ul><li><strong>Output Directory (<code>OUT_DIR</code>)</strong>: The path where the script saves the trimmed files. Modify this to suit your project structure.</li><li><strong>Sample Prefixes (<code>Prefixes</code> Array)</strong>: This array holds the identifiers for the sample sets to be processed. These identifiers are crucial for generating the filenames for paired-end reads.</li><li><strong>Processing Loop</strong>: For each sample identifier in the <code>Prefixes</code> array, the script constructs the filenames for R1 and R2 reads and executes Trim Galore to trim these files.</li><li><strong>Trim Galore Command</strong>: The script runs Trim Galore with parameters set for quality (<code>-q 25</code>), encoding (<code>--phred33</code>), stringency (<code>--stringency 3</code>), minimum length (<code>--length 30</code>), and output compression (<code>--gzip</code>), directing the output to the specified directory.</li></ul><p><strong>To run this script:</strong></p><ol><li>Save the script into a file, e.g., <code>trim_script.sh</code>.</li><li>Make the script executable by running <code>chmod +x trim_script.sh</code>.</li><li>Execute the script with <code>./trim_script.sh</code> in your terminal.</li></ol><p>This script aims to streamline the preprocessing of multiple RNA-seq samples by automating the trimming process, ensuring consistency and saving valuable time during quality control checks.</p><h2 id=4-quality-check-post-trimming-with-fastqc--multiqc>4. Quality Check Post-Trimming with <code>FastQC</code> & <code>MultiQC</code></h2><p>It&rsquo;s crucial to reassess the quality of your reads after trimming.</p><div class=highlight><pre tabindex=0 style=color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4><code class=language-shell data-lang=shell><span style=display:flex><span>fastqc *.gz
</span></span><span style=display:flex><span>multiqc .
</span></span></code></pre></div><img src=https://s2.loli.net/2022/04/08/QJoiqIPwBV3jg6b.png style=zoom:90%><p>The quality significantly improves after trimming! :v:</p></div><div class="row items-start justify-between"><div class="lg:col-5 mb-10 flex items-center lg:mb-0"><h5 class=mr-3>Tags :</h5><ul><li class=inline-block><a class="bg-theme-light hover:bg-primary m-1 block rounded px-3 py-1 hover:text-white" href=/tags/rna-seq/>Rna seq</a></li><li class=inline-block><a class="bg-theme-light hover:bg-primary m-1 block rounded px-3 py-1 hover:text-white" href=/tags/quality-control/>Quality control</a></li></ul></div><div class="lg:col-4 flex items-center"><div class=share-icons><h5 class=share-title>Share :</h5><a class="share-link share-facebook" href="https://facebook.com/sharer/sharer.php?u=%2fblog%2ftrim-the-read-files-using-trim_galore%2f" target=_blank rel=noopener aria-label="share facebook"><span class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18.77 7.46H14.5v-1.9c0-.9.6-1.1 1-1.1h3V.5h-4.33C10.24.5 9.5 3.44 9.5 5.32v2.15h-3v4h3v12h5v-12h3.85l.42-4z"/></svg>
</span></a><a class="share-link share-twitter" href="https://twitter.com/intent/tweet/?text=Share&amp;url=%2fblog%2ftrim-the-read-files-using-trim_galore%2f" target=_blank rel=noopener aria-label="share twitter"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8 2H1l8.26 11.015L1.45 22H4.1l6.388-7.349L16 22h7l-8.608-11.478L21.8 2h-2.65l-5.986 6.886zm9 18L5 4h2l12 16z"/></svg>
</span></a><a class="share-link share-email" href="mailto:?subject=Share&amp;body=%2fblog%2ftrim-the-read-files-using-trim_galore%2f" target=_self rel=noopener aria-label="share email"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M22 4H2C.9 4 0 4.9.0 6v12c0 1.1.9 2 2 2h20c1.1.0 2-.9 2-2V6c0-1.1-.9-2-2-2zM7.25 14.43l-3.5 2c-.08.05-.17.07-.25.07-.17.0-.34-.1-.43-.25-.14-.24-.06-.55.18-.68l3.5-2c.24-.14.55-.06.68.18.14.24.06.55-.18.68zm4.75.07c-.1.0-.2-.03-.27-.08l-8.5-5.5c-.23-.15-.3-.46-.15-.7.15-.22.46-.3.7-.14L12 13.4l8.23-5.32c.23-.15.54-.08.7.15.14.23.07.54-.16.7l-8.5 5.5c-.08.04-.17.07-.27.07zm8.93 1.75c-.1.16-.26.25-.43.25-.08.0-.17-.02-.25-.07l-3.5-2c-.24-.13-.32-.44-.18-.68s.44-.32.68-.18l3.5 2c.24.13.32.44.18.68z"/></svg>
</span></a><a class="share-link share-reddit" href="https://reddit.com/submit/?url=%2fblog%2ftrim-the-read-files-using-trim_galore%2f&amp;resubmit=true&amp;title=Share" target=_blank rel=noopener aria-label="share reddit"><span aria-hidden=true class=share-icon><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M24 11.5c0-1.65-1.35-3-3-3-.96.0-1.86.48-2.42 1.24-1.64-1-3.75-1.64-6.07-1.72.08-1.1.4-3.05 1.52-3.7.72-.4 1.73-.24 3 .5C17.2 6.3 18.46 7.5 20 7.5c1.65.0 3-1.35 3-3s-1.35-3-3-3c-1.38.0-2.54.94-2.88 2.22-1.43-.72-2.64-.8-3.6-.25-1.64.94-1.95 3.47-2 4.55-2.33.08-4.45.7-6.1 1.72C4.86 8.98 3.96 8.5 3 8.5c-1.65.0-3 1.35-3 3 0 1.32.84 2.44 2.05 2.84-.03.22-.05.44-.05.66.0 3.86 4.5 7 10 7s10-3.14 10-7c0-.22-.02-.44-.05-.66 1.2-.4 2.05-1.54 2.05-2.84zM2.3 13.37C1.5 13.07 1 12.35 1 11.5c0-1.1.9-2 2-2 .64.0 1.22.32 1.6.82-1.1.85-1.92 1.9-2.3 3.05zm3.7.13c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2zm9.8 4.8c-1.08.63-2.42.96-3.8.96-1.4.0-2.74-.34-3.8-.95-.24-.13-.32-.44-.2-.68.15-.24.46-.32.7-.18 1.83 1.06 4.76 1.06 6.6.0.23-.13.53-.05.67.2.14.23.06.54-.18.67zm.2-2.8c-1.1.0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm5.7-2.13c-.38-1.16-1.2-2.2-2.3-3.05.38-.5.97-.82 1.6-.82 1.1.0 2 .9 2 2 0 .84-.53 1.57-1.3 1.87z"/></svg></span></a></div></div></div><div class=mt-20><div id=disqus_thread></div><script>window.disqus_config=function(){},function(){if(["localhost","127.0.0.1"].indexOf(window.location.hostname)!=-1){document.getElementById("disqus_thread").innerHTML="Disqus comments not available by default when the website is previewed locally.";return}var t=document,e=t.createElement("script");e.async=!0,e.src="//themefisher-template.disqus.com/embed.js",e.setAttribute("data-timestamp",+new Date),(t.head||t.body).appendChild(e)}()</script><noscript>Please enable JavaScript to view the <a href=https://disqus.com/?ref_noscript>comments powered by Disqus.</a></noscript><a href=https://disqus.com class=dsq-brlink>comments powered by <span class=logo-disqus>Disqus</span></a></div></article></div><div class="section pb-0"><h2 class="h3 mb-12">Related Posts</h2><div class=row><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/PCA_hu87be8c1307d3013bc46fdca9c9c0afca_306677_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="Bulk RNA-seq (6): Normalization and PCA Visualization in RNA-seq Data Analysis with DESeq2" width=9448 height=4724></picture><h4 class=mb-3><a href=/blog/post-2/>Bulk RNA-seq (6): Normalization and PCA Visualization in RNA-seq Data Analysis with DESeq2</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>Note
In this blog post, we&rsquo;ll dive into a practical guide on using DESeq2 for differential expression analysis in RNA-seq data.</p><a class="btn btn-outline-primary btn-sm" href=/blog/post-2/>Read More</a></div></div><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/mappingscripts_hu66a195306e7d03dd8abb316f1d6fa6b6_637620_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="Bulk RNA-seq (5): Streamlining Mapping with a Custom Linux Script" width=6000 height=3600></picture><h4 class=mb-3><a href=/blog/streamlining-mapping-with-a-custom-linux-script/>Bulk RNA-seq (5): Streamlining Mapping with a Custom Linux Script</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>Note
To overcome the inconvenience of manually mapping each RNA-seq sample to the reference genome, I&rsquo;ve developed a Linux shell script.</p><a class="btn btn-outline-primary btn-sm" href=/blog/streamlining-mapping-with-a-custom-linux-script/>Read More</a></div></div><div class=lg:col-4><div class=bg-body><picture><source srcset=/images/gGtLYrNR1af7jxW_hu7b7071b4195e7df5666eecdb10736f3a_449380_545x0_resize_q100_h2_lanczos_3.webp media="(max-width: 575px)"><source srcset=/images/gGtLYrNR1af7jxW_hu7b7071b4195e7df5666eecdb10736f3a_449380_600x0_resize_q100_h2_lanczos_3.webp media="(max-width: 767px)"><source srcset=/images/gGtLYrNR1af7jxW_hu7b7071b4195e7df5666eecdb10736f3a_449380_700x0_resize_q100_h2_lanczos_3.webp media="(max-width: 991px)"><source srcset=/images/gGtLYrNR1af7jxW_hu7b7071b4195e7df5666eecdb10736f3a_449380_1110x0_resize_q100_h2_lanczos_3.webp><img loading=lazy decoding=async src=/images/gGtLYrNR1af7jxW_hu7b7071b4195e7df5666eecdb10736f3a_449380_1110x0_resize_lanczos_3.png class="mb-6 w-full rounded img" alt="Bulk RNA-seq (4): Mapping and Quality Control with STAR, Qualimap, and featureCounts" width=1572 height=793></picture><h4 class=mb-3><a href=/blog/mapping-and-quality-control-with-star-qualimap-and-featurecounts/>Bulk RNA-seq (4): Mapping and Quality Control with STAR, Qualimap, and featureCounts</a></h4><ul class=mb-4><li class="mr-4 inline-block"><a href=/authors/yuan-yao/><i class="fa-regular fa-circle-user mr-2"></i>Yuan Yao</a></li><li class="mr-4 inline-block"><i class="fa-regular fa-folder mr-1"></i>
<a href=/categories/rna-seq/ class=ms-1>Rna seq</a></li></ul><p class=mb-6>Note
This section of the blog series will guide you through mapping RNA-seq read files to a genome index using STAR, performing quality control (QC) with Qualimap, and generating count data with featureCounts.</p><a class="btn btn-outline-primary btn-sm" href=/blog/mapping-and-quality-control-with-star-qualimap-and-featurecounts/>Read More</a></div></div></div></div></div></section></main><footer class=bg-theme-light><div class=container><div class="row items-center py-10"><div class="lg:col-3 mb-8 text-center lg:mb-0 lg:text-left"><a class="navbar-brand inline-block" href=/><img fetchpriority=high decoding=async class="img logo-light" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'>
<img fetchpriority=high decoding=async class="img logo-dark" width=200 height=100 src=/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_q100_h2_lanczos_3.webp alt=YAO onerror='this.onerror=null,this.src="/images/logo1_hud0c72c687cfa3f3c58982b839d7f7043_93760_400x0_resize_lanczos_3.png"'></a></div><div class="lg:col-6 mb-8 text-center lg:mb-0"><ul><li class="m-3 inline-block"><a href=/about/>About</a></li><li class="m-3 inline-block"><a href=/elements/>Elements</a></li><li class="m-3 inline-block"><a href=/privacy-policy/>Privacy Policy</a></li></ul></div><div class="lg:col-3 mb-8 text-center lg:mb-0 lg:mt-0 lg:text-right"><ul class=social-icons><li><a target=_blank aria-label=facebook rel="nofollow noopener" href=https://www.facebook.com/><i class="fab fa-facebook"></i></a></li><li><a target=_blank aria-label=twitter rel="nofollow noopener" href=https://twitter.com/><i class="fab fa-twitter"></i></a></li><li><a target=_blank aria-label=github rel="nofollow noopener" href=https://www.github.com/><i class="fab fa-github"></i></a></li><li><a target=_blank aria-label=linkedin rel="nofollow noopener" href=https://www.linkedin.com/><i class="fab fa-linkedin"></i></a></li></ul></div></div></div><div class="border-border border-t py-7"><div class="text-light container text-center"><p>Designed & Developed by Yao Yuan</p></div></div></footer><script crossorigin=anonymous integrity="sha256-EjMwAhXn+vYO0J+UkUIy8BjNtBx1PoMmC+zQvUyDVsY=" src=/js/script.min.1233300215e7faf60ed09f94914232f018cdb41c753e83260becd0bd4c8356c6.js></script><script defer async crossorigin=anonymous integrity="sha256-9e9iyznGCSfFZB4ElW+WFonrJpHJr5pYxsaD7VVh4wc=" src=/js/script-lazy.min.f5ef62cb39c60927c5641e04956f961689eb2691c9af9a58c6c683ed5561e307.js></script><script>"serviceWorker"in navigator&&navigator.serviceWorker.register("/service-worker.js")</script></body></html>